# Generated by Django 3.1.14 on 2022-08-23 16:18
import os

from django.db import migrations, IntegrityError
from django.contrib.auth.models import User


def generate_superuser(*args, **kwargs):
    username = os.environ.get("DJANGO_SUPERUSER_USERNAME")
    password = os.environ.get("DJANGO_SUPERUSER_PASSWORD")
    email = os.environ.get("DJANGO_SUPERUSER_EMAIL")
    try:
        superuser = User.objects.create_superuser(
            username=username,
            email=email,
            password=password
        )
        superuser.save()
    except IntegrityError:
        print("admin user already exists in db")


class Migration(migrations.Migration):

    dependencies = [
        ('viewer', '0012_auto_20220808_1857'),
    ]

    operations = [
        migrations.RunPython(generate_superuser),
    ]
